graspologic/embed/ase.py

```diff
@@ -158,4 +160,52 @@ def fit(self, graph, y=None):
 
         self.n_features_in_ = len(A)
         self._reduce_dim(A)
+
+        # for out-of-sample
+        inv_eigs = np.diag(1 / self.singular_values_)
+        self.pinv_left_ = self.latent_left_ @ inv_eigs
+        if self.latent_right_ is not None:
+            self.pinv_right_ = self.latent_right_ @ inv_eigs
+
+        self.is_fitted_ = True
         return self
+
+    def transform(self, y):
+        """
+        Obtain an out-of-sample embedding from a vertex not in the original embedding.
+        For more details, see [1].
+
+        Parameters
+        ----------
+        y : array_like or tuple, shape (n_oos_samples, n_vertices)
+            out-of-sample matrix.
+            If tuple, graph is directed and y[0] contains edges from y to other nodes.
+
+        Returns
+        -------
+        array_like or tuple
+            Out-of-sample prediction for the latent position(s) of y.
+            If the original embedding was undirected, input should be an array.
+            If the original embedding was directed, input should be a tuple (y_right, y_left).
+
+        References
+        ----------
+        .. [1] Levin, K., Roosta-Khorasani, F., Mahoney, M. W., & Priebe, C. E. (2018).
+        Out-of-sample extension of graph adjacency spectral embedding. PMLR: Proceedings
+        of Machine Learning Research, 80, 2975-2984.
+        """
+
+        # checks
+        check_is_fitted(self, "is_fitted_")
+        if not np.array_equal(y, y.astype(bool)):
+            raise ValueError("Out-of-sample array must be unweighted.")
```

Comment:
say what?

Reply:
```
In [1]: import numpy as np

In [2]: a = np.array([1, 0, 0])

In [3]: a.astype(bool)
Out[3]: array([ True, False, False])

In [4]: np.array_equal(a, a.astype(bool))
Out[4]: True

In [5]: b = a + 1

In [6]: b
Out[6]: array([2, 1, 1])

In [7]: np.array_equal(b, b.astype(bool))
Out[7]: False

In [8]: c = np.array([2, 0, 0])

In [9]: c.astype(bool)
Out[9]: array([ True, False, False])

In [10]: np.array_equal(c, c.astype(bool))
Out[10]: False
```

Reply:
Not what I meant, the line I tried to comment on was 
`Out-of-sample array must be unweighted.`
why?

Reply:
oh, good point. I put that in originally cause I hadn't tested on weighted graphs yet. I'll remove it once I play around with a weighted graph and make sure nothing is funky (although I don't see why it would be)

Reply:
ya, weighted is most common use case, i think we wanna test/support that. 

one thing that could get funky is `pass_to_ranks()` which is most common preprocessing step we use for weighted graphs. perhaps out of scope for this PR but will be worth thinking about how to due a true OOS embedding without knowledge of full graph but still using PTR.

Reply:
anyway I'll shut up till peer review happens :)